### ISSUE TITLE
Some questions about iou loss and other variants

### ISSUE THREAD

**Issue by wangjue-wzq** [open-mmlab/mmdetection#4454]
I use giou loss，iou loss and diou loss. Like this:
IoULoss:
`

bbox_head=dict(
        type='SharedFCBBoxHead',
        num_fcs=2,         #
        in_channels=256,
        fc_out_channels=1024,   #
        roi_feat_size=7,
        num_classes=4,     #number classes + 1
        target_means=[0., 0., 0., 0.],
        target_stds=[0.1, 0.1, 0.2, 0.2],
        reg_class_agnostic=False,
        loss_cls=dict(
            type='CrossEntropyLoss', use_sigmoid=False, loss_weight=1.0),
        loss_bbox=dict(type='IoULoss', loss_weight=1.0)))

`
GIoULoss
`

bbox_head=dict(
        type='SharedFCBBoxHead',
        num_fcs=2,         #
        in_channels=256,
        fc_out_channels=1024,   #
        roi_feat_size=7,
        num_classes=4,     #number classes + 1
        target_means=[0., 0., 0., 0.],
        target_stds=[0.1, 0.1, 0.2, 0.2],
        reg_class_agnostic=False,
        loss_cls=dict(
            type='CrossEntropyLoss', use_sigmoid=False, loss_weight=1.0),
        loss_bbox=dict(type='GIoULoss', loss_weight=1.0)))

`
DIoULoss
`

bbox_head=dict(
        type='SharedFCBBoxHead',
        num_fcs=2,         #
        in_channels=256,
        fc_out_channels=1024,   #
        roi_feat_size=7,
        num_classes=4,     #number classes + 1
        target_means=[0., 0., 0., 0.],
        target_stds=[0.1, 0.1, 0.2, 0.2],
        reg_class_agnostic=False,
        loss_cls=dict(
            type='CrossEntropyLoss', use_sigmoid=False, loss_weight=1.0),
        loss_bbox=dict(type='DIoULoss', loss_weight=1.0)))

`
When I use DIoULoss and IoULoss, the mAP is very poor, about 30. When I use GIoU the loss is NAN after some epochs like this:
`

{"mode": "train", "epoch": 9, "iter": 1400, "lr": 0.01, "time": 0.53069, "data_time": 0.00923, "memory": 7000, "loss_rpn_cls": 0.01162, "loss_rpn_bbox": 0.01638, "loss_cls": 0.13627, "acc": 94.13281, "loss_small_cls": 0.03073, "loss_bbox": 0.13018, "loss_small_bbox": 0.06462, "loss": 0.3898}
{"mode": "train", "epoch": 9, "iter": 1450, "lr": 0.01, "time": 0.52851, "data_time": 0.00943, "memory": 7000, "loss_rpn_cls": 0.01003, "loss_rpn_bbox": 0.01501, "loss_cls": 0.13183, "acc": 94.33789, "loss_small_cls": 0.02948, "loss_bbox": 0.13219, "loss_small_bbox": 0.05705, "loss": 0.37559}
{"mode": "train", "epoch": 9, "iter": 1500, "lr": 0.01, "time": 0.53235, "data_time": 0.0093, "memory": 7000, "loss_rpn_cls": 0.01183, "loss_rpn_bbox": 0.01733, "loss_cls": 0.13184, "acc": 94.52246, "loss_small_cls": 0.03295, "loss_bbox": 0.13757, "loss_small_bbox": 0.06812, "loss": 0.39964}
{"mode": "train", "epoch": 9, "iter": 1550, "lr": 0.01, "time": 0.51412, "data_time": 0.00917, "memory": 7000, "loss_rpn_cls": 4256643138353235.0, "loss_rpn_bbox": 288576570518847.6, "loss_cls": NaN, "acc": 69.95814, "loss_small_cls": NaN, "loss_bbox": NaN, "loss_small_bbox": NaN, "loss": NaN}
{"mode": "train", "epoch": 9, "iter": 1600, "lr": 0.01, "time": 0.50229, "data_time": 0.00901, "memory": 7000, "loss_rpn_cls": 0.60825, "loss_rpn_bbox": 0.04782, "loss_cls": NaN, "acc": 31.54492, "loss_small_cls": NaN, "loss_bbox": NaN, "loss_small_bbox": NaN, "loss": NaN}
{"mode": "train", "epoch": 9, "iter": 1650, "lr": 0.01, "time": 0.5066, "data_time": 0.00928, "memory": 7000, "loss_rpn_cls": 0.47215, "loss_rpn_bbox": 0.05039, "loss_cls": NaN, "acc": 29.94141, "loss_small_cls": NaN, "loss_bbox": NaN, "loss_small_bbox": NaN, "loss": NaN}
{"mode": "train", "epoch": 9, "iter": 1700, "lr": 0.01, "time": 0.51834, "data_time": 0.00945, "memory": 7000, "loss_rpn_cls": 0.4091, "loss_rpn_bbox": 0.04851, "loss_cls": NaN, "acc": 29.95605, "loss_small_cls": NaN, "loss_bbox": NaN, "loss_small_bbox": NaN, "loss": NaN}
{"mode": "train", "epoch": 9, "iter": 1750, "lr": 0.01, "time": 0.5093, "data_time": 0.00959, "memory": 7000, "loss_rpn_cls": 0.38699, "loss_rpn_bbox": 0.04748, "loss_cls": NaN, "acc": 35.38379, "loss_small_cls": NaN, "loss_bbox": NaN, "loss_small_bbox": NaN, "loss": NaN}
{"mode": "train", "epoch": 9, "iter": 1800, "lr": 0.01, "time": 0.50763, "data_time": 0.00911, "memory": 7000, "loss_rpn_cls": 0.38594, "loss_rpn_bbox": 0.05837, "loss_cls": NaN, "acc": 42.22754, "loss_small_cls": NaN, "loss_bbox": NaN, "loss_small_bbox": NaN, "loss": NaN}
{"mode": "train", "epoch": 9, "iter": 1850, "lr": 0.01, "time": 0.50576, "data_time": 0.00959, "memory": 7000, "loss_rpn_cls": 0.35516, "loss_rpn_bbox": 0.04771, "loss_cls": NaN, "acc": 37.63379, "loss_small_cls": NaN, "loss_bbox": NaN, "loss_small_bbox": NaN, "loss": NaN}
{"mode": "train", "epoch": 9, "iter": 1900, "lr": 0.01, "time": 0.50817, "data_time": 0.0091, "memory": 7000, "loss_rpn_cls": 0.34512, "loss_rpn_bbox": 0.04973, "loss_cls": NaN, "acc": 25.51562, "loss_small_cls": NaN, "loss_bbox": NaN, "loss_small_bbox": NaN, "loss": NaN}
{"mode": "train", "epoch": 9, "iter": 1950, "lr": 0.01, "time": 0.51296, "data_time": 0.00938, "memory": 7000, "loss_rpn_cls": 0.34682, "loss_rpn_bbox": 0.05066, "loss_cls": NaN, "acc": 37.29102, "loss_small_cls": NaN, "loss_bbox": NaN, "loss_small_bbox": NaN, "loss": NaN}
{"mode": "train", "epoch": 9, "iter": 2000, "lr": 0.01, "time": 0.50756, "data_time": 0.00928, "memory": 7000, "loss_rpn_cls": 0.30973, "loss_rpn_bbox": 0.04185, "loss_cls": NaN, "acc": 29.27148, "loss_small_cls": NaN, "loss_bbox": NaN, "loss_small_bbox": NaN, "loss": NaN}

`
But when I use SmoothL1Loss, the mAP is normal about 90. Can you give me some suggestions?

**Comment by ZwwWayne** [comment]
You may need to set `reg_decoded_bbox=True` in the config of bboxhead.

**Comment by wangjue-wzq** [comment]
Ok! Thank you! I will try.

**Comment by ZwwWayne** [comment]
Resolved in #4467.
